packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -105,6 +113,13 @@ export class EthereumAnchorService implements AnchorService {
     return [this._chainId]
   }
 
+  private _carFileFromRequestAnchorParams(params: RequestAnchorParams): CAR {
+    const carFactory = new CARFactory()
+    const car = carFactory.build()
+    car.put({...params})
```

Comment:
This doesn't look right to me. We should put each individual IPLD object into the car file. 

Should include:
* genesisCommit 
  * genesisCommit JWS (if signed)
  * genesisCommit CACAO (if created with session key)
* Actual commit:
  * Data object
  * JWS
  * CACAO (if present)
* metadata object, including the following info:
  * timestamp
  * streamtype number (can be used with genesis commit to create streamid

Then I'm not sure if maybe we want to sign the metadata object? Anyway, the metadata object should be set as `isRoot`.

Reply:
Thanks, @oed !

With @ukstv 's help, I fixed the structure of the CAR file.

Still not passing the actual commit, though, because @ukstv  and me agreed that:
* we wouldn't be using it for verification on CAS side in the foreseeable future
* when we do decide to check the signatures of the whole stream when creating the merkle tree, in general case would would most likely have to send more than just the anchored commit anyway. 

So it's gonna be best just to sent the genesis commit for now and then design the next step properly. Does this make sense, @oed ?

When it comes to signing the metadata, as a part of the CAS Auth work, @v-stickykeys has implemented signing of the whole request body for authenticated requests and we agreed that this will be good enough in the current stage, although we might want to optimise it for better performance in the future. WDYT, @oed ?

Reply:
Yes, integration with @v-stickykeys work makes sense. I'm not sure I follow the logic of not including the signed commit and the associated payload and CACAOs though? I thought this was spec'd out already?

Reply:
> why do you think we need to send the actual commit and a CACAO as well

They are needed to verify the signature of the commit on CAS. Without this signature verification a Ceramic node could spam CAS with a lot of fake commits for a stream, making historical indexing super slow because now syncing nodes would need to load a lot of commits.



It's worth noting that the genesis commit might be one ipld object (which you support in your code). It may also be a data object, a DagJWS object, and a CACAO object in the case of a signed genesis.

Reply:
> They are needed to verify the signature of the commit on CAS. 

Ok, let me sync with @ukstv and see, if this signature verification had been planned for this project and if not, how much additional time it would take.

On a side note, is it sufficient to send just the last commit in an anchor request, if CAS is supposed to verify it's signature? Won't CAS need the whole stream?

> It's worth noting that the genesis commit might be one ipld object (which you support in your code). It may also be a data object, a DagJWS object, and a CACAO object in the case of a signed genesis.

This is how I get the genesis payload before putting it in the CAR object: `const genesisPayload = await this.dispatcher._ipfs.block.get(state$.value.log[0].cid)`. Will this only give me unsigned genesis commits? Should I retrieve signed genesis commits in a different way?

Reply:
> if this signature verification had been planned for this project and if not, how much additional time it would take.

I'm fine with adding this later as long as we are aware that it would be a quite high priority project.

For future compatibility it would be nice if we sent along the entire commit being anchored, so that CAS could start verifying signatures at some point in the future!

> On a side note, is it sufficient to send just the last commit in an anchor request, if CAS is supposed to verify it's signature? Won't CAS need the whole stream?

If we make the assumption that the stream doesn't allow controller changes (e.g. MIDs and caip10-links) then we won't need the entire stream. If we do allow controller changes (like in TileDocuments) then yes the entire stream would need to be loaded. A tradeoff here could be to verify signatures for all streamtypes except Tiles.



 > Will this only give me unsigned genesis commits? Should I retrieve signed genesis commits in a different way?

This will either give you an unsigned genesis commit, or it will give you the DagJWS object. You can detect this if the loaded data has a `link` field.

Reply:
Ok, @oed , thanks for the responses :). I'm gonna add the anchored commit to the CAR file straight away and then we'll aim at adding the verification within this project or right after.